src/charm.py

```diff
@@ -165,14 +165,21 @@ def _on_config_changed(self, event: ConfigChangedEvent) -> None:
             )
             event.defer()
             return
-        if self._certificates_relation_created:
-            self._revoke_existing_certificates()
-            self._generate_csr()
-            self._request_certificate()
-            self.unit.status = WaitingStatus("Waiting for certificate to be available")
-        else:
-            self._generate_csr()
-            self.unit.status = ActiveStatus()
+        if not self._certificates_relation_created:
```

Comment:
I would disagree with this order.
This will push the user to relate the application before the csr is first generated.

Reply:
I think the previous order was okay, only status management had to be improved.

Reply:
On the other hand what would be the point of creating CSR without having certs provider integrated?

Reply:
I tend to support @Gmerold's opinion. Without having relating with a provider, creating csr seems useless. Could we discuss this in daily ?